packages/accounts/src/msca/builder.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  BaseSmartContractAccount,
+  type BaseSmartAccountParams,
+  type BatchUserOperationCallData,
+  type ISmartContractAccount,
+  type SignTypedDataParams,
+  type SupportedTransports,
+} from "@alchemy/aa-core";
+import {
+  encodeFunctionData,
+  type Address,
+  type Hex,
+  type Transport,
+} from "viem";
+import { z } from "zod";
+import { IStandardExecutorAbi } from "./abis/IStandardExecutor.js";
+import type { Plugin } from "./plugins/types";
+
+export interface MSCA extends ISmartContractAccount {
+  extendWithPluginMethods: <D>(plugin: Plugin<D>) => this & D;
+}
+
+export type Executor = <A extends MSCA>(
+  acct: A
+) => Pick<ISmartContractAccount, "encodeExecute" | "encodeBatchExecute">;
+
+export type SignerMethods = <A extends MSCA>(
+  acct: A
+) => Pick<
+  ISmartContractAccount,
+  | "signMessage"
+  | "signTypedData"
+  | "signUserOperationHash"
+  | "getDummySignature"
```

Comment:
these are abstraction functions except the execute functions from SmartContractAccount correct?

Then why also `signUserOperationHash`? Don't we just use `signMessage` for signing uos?

Reply:
for 1271 validation, we sign messages differently that we do for UO hashes. The MSCA (and LightAccount) expects the UO to be personal sign of the hash. 1271 expects a sign typed data following the envelope defined above